website/data-storage/src/messages/messages.service.ts

```diff
@@ -121,27 +122,12 @@ export class MessagesService {
     return result;
   }
 
-  async setDispatchedStatus(params: MessageDispatched): Promise<void> {
-    const error = params.outcome !== 'success' ? params.outcome : null;
-    if (error === null) {
-      return;
-    }
-    await sleep(1000);
-    const message = await this.messageRepo.findOne({
-      where: { genesis: params.genesis, id: params.messageId },
-    });
-    if (message) {
-      message.error = error;
-      this.messageRepo.save(message);
-    }
-    const logMessages = await this.messageRepo.find({
-      where: { genesis: params.genesis, replyTo: params.messageId, replyError: '1' },
-    });
-    if (logMessages.length > 0) {
-      logMessages.forEach((log) => {
-        log.replyError = error;
-        this.messageRepo.save(log);
-      });
+  async setDispatchedStatus({ statuses, genesis }: IMessagesDispatchedKafkaValue): Promise<void> {
+    for (let messageId of Object.keys(statuses)) {
+      if (statuses[messageId] === 'Success') continue;
+
+      await sleep(100);
```

Comment:
Simple refactor. Call two independent features.

```javascript
Promise.all([
sleep(10),
this.messageRepo.update({ genesis, id: messageId }, { processedWithPanic: true })
])
```

Reply:
No, these aren't independent. sleep() is necessary, because sometimes it happens that this method is called before the message that needs to be update is saved in database

Reply:
Alright. I got it.